src/event/utils.ts

```diff
@@ -57,6 +81,23 @@ async function _callHandler(event: H3Event, handler: EventHandlerObject) {
       }
     }
   }
+  if (handler.validate) {
+    const res = (await validateEvent(event, handler.validate)) as Record<
+      string,
+      any
+    >;
+    if (res && typeof res === "object") {
+      for (const property in res) {
+        if (property === "body") {
+          (event.node.req as any)[ParsedBodySymbol] = res.body;
```

Comment:
We might keep readBody utils decoupled from this new feature of context injection. (the read options could be different)